crates/node/src/extensions/pragma_dispatch.rs

```diff
@@ -1,15 +1,92 @@
 //! ExEx of Pragma Dispatcher
 //! Adds a new TX at the end of each block, dispatching a message through
 //! Hyperlane.
+use std::sync::Arc;
 
 use futures::StreamExt;
-use mp_exex::{ExExContext, ExExEvent};
+use starknet_api::felt;
+use starknet_core::types::{
+    BroadcastedInvokeTransaction, BroadcastedInvokeTransactionV1, BroadcastedTransaction, Felt,
+};
+use starknet_signers::SigningKey;
 
+use mc_devnet::{Call, Multicall, Selector};
+use mc_mempool::transaction_hash;
+use mp_chain_config::ChainConfig;
+use mp_convert::ToFelt;
+use mp_exex::{ExExContext, ExExEvent, ExExNotification};
+use mp_transactions::broadcasted_to_blockifier;
+
+lazy_static::lazy_static! {
+    pub static ref ACCOUNT_ADDRESS: Felt = felt!("0x4a2b383d808b7285cc98b2309f974f5111633c84fd82c9375c118485d2d57ba");
+    pub static ref PRIVATE_KEY: SigningKey = SigningKey::from_secret_scalar(felt!("0x7a9779748888c95d96bbbce041b5109c6ffc0c4f30561c0170384a5922d9e91"));
+
+    pub static ref PRAGMA_DISPATCHER_ADDRESS: Felt = felt!("0x2a85bd616f912537c50a49a4076db02c00b29b2cdc8a197ce92ed1837fa875b");
+    pub static ref PRAGMA_FEED_IDS: Vec<Felt> = vec![
+        felt!("18669995996566340"), // BTC/USD: Spot Median
+        felt!("19514442401534788"), // ETH/USD: Spot Median
+    ];
+}
+
+/// 🧩 Pragma main ExEx.
+/// At the end of each produced block by the node, adds a new dispatch transaction
+/// using the Pragma Dispatcher contract.
 pub async fn exex_pragma_dispatch(mut ctx: ExExContext) -> anyhow::Result<()> {
+    let mut nonce = Felt::ZERO;
     while let Some(notification) = ctx.notifications.next().await {
-        let block_number = notification.closed_block();
-        log::info!("👋 Hello from the ExEx (triggered at block #{})", block_number);
+        let block_number = match notification {
+            ExExNotification::BlockProduced { block: _, block_number } => block_number,
```

Comment:
why would we differentiate between `BlockProduced` and `BlockSynced` tho, it's the same meaning just on a different pov no?

Reply:
Yup that's what I wanted at first but:
* I think ExExs should be able to choose depending on the node mode if they're ran or not. This should be done in a higher level, but for example it does not make that much sense for our ExEx to have the same behavior in a sync mode than in a sequencer mode
* if we want to pass the Block that has been produced/synced to the notification, since the types are different (PreValidatedBlock / MadaraPendingBlock) we would need some new trait

So probably the best solution here is:
1. ExExs should have some sort of configs saying "I will be only ran in sequencer/full/both mode"
2. Some kind of ̀`ClosedBlock` trait implemented over both structs & passed to the notification

Reply:
lets keep that in mind and open a discussion in madara repo, I think these are valid points but there are different designs possible